// styled-alert-component/lib/components/Alert.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Alert = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styledBaseComponents = require("styled-base-components");

var _styledConfig = require("styled-config");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _templateObject10() {
  var data = _taggedTemplateLiteral(["\n  position: relative;\n  padding: ", ";\n  border: ", ";\n  & > a {\n    font-weight: ", ";\n  };\n  & > h1,\n  & > h2,\n  & > h3,\n  & > h4,\n  & > h5,\n  & > h6 {\n    color: inherit;\n  };\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteral(["border-radius: ", ""]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["border-radius: ", ""]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["border-radius: ", ""]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n    & > a,\n    & > a:hover {\n      color: ", ";\n    };\n  "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n    & > hr {\n      border-top-color: ", ";\n    };\n  "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["border-color: ", ";"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["background-color: ", ";"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["color: ", ";"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    padding-right: ", ";\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var alertDismissible = function alertDismissible(props) {
  return props.dismissible && (0, _styledComponents.css)(_templateObject(), (0, _styledConfig.getPadding)(props, 'alert', 'dismissibleRight'));
};

var color = function color(props) {
  return (0, _styledComponents.css)(_templateObject2(), (0, _styledConfig.getColor)(props, 'alert', 'text'));
};

var backgroundColor = function backgroundColor(props) {
  return (0, _styledComponents.css)(_templateObject3(), (0, _styledConfig.getColor)(props, 'alert', 'background'));
};

var borderColor = function borderColor(props) {
  return (0, _styledComponents.css)(_templateObject4(), (0, _styledConfig.getColor)(props, 'alert', 'border'));
};

var borderTopColor = function borderTopColor(props) {
  return (0, _styledComponents.css)(_templateObject5(), (0, _styledConfig.getColor)(props, 'alert', 'borderTop'));
};

var linkHoverColor = function linkHoverColor(props) {
  return (0, _styledComponents.css)(_templateObject6(), (0, _styledConfig.getColor)(props, 'alert', 'hover'));
};

var borderRadius = function borderRadius(props) {
  if (props.noRadius) {
    return (0, _styledComponents.css)(_templateObject7(), (0, _styledConfig.getBorderRadius)(props, 'alert', 'noRadius'));
  } else if (props.pill) {
    return (0, _styledComponents.css)(_templateObject8(), (0, _styledConfig.getBorderRadius)(props, 'alert', 'pill'));
  }

  return (0, _styledComponents.css)(_templateObject9(), (0, _styledConfig.getBorderRadius)(props, 'alert', 'default'));
};

var Alert = (0, _styledComponents.default)(_styledBaseComponents.Div)(_templateObject10(), function (props) {
  return (0, _styledConfig.getPadding)(props, 'alert', 'default');
}, function (props) {
  return (0, _styledConfig.getBorder)(props, 'alert', 'default');
}, function (props) {
  return (0, _styledConfig.getFontWeight)(props, 'alert', 'default');
}, function (props) {
  return borderRadius(props);
}, function (props) {
  return alertDismissible(props);
}, function (props) {
  return color(props);
}, function (props) {
  return backgroundColor(props);
}, function (props) {
  return borderColor(props);
}, function (props) {
  return borderTopColor(props);
}, function (props) {
  return linkHoverColor(props);
});
exports.Alert = Alert;
Alert.defaultProps = {
  theme: _styledConfig.theme
};
