// styled-modal-component/lib/components/Modal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Modal = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledBaseComponents = require("styled-base-components");

var _styledConfig = require("styled-config");

var _ModalContent = require("./ModalContent");

var _ModalDialog = require("./ModalDialog");

var _ModalBackdrop = require("./ModalBackdrop");

var _ModalWrapper = require("./ModalWrapper");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

var Modal =
/*#__PURE__*/
function (_React$Component) {
  _inherits(Modal, _React$Component);

  function Modal(props) {
    var _this;

    _classCallCheck(this, Modal);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(Modal).call(this, props));
    _this.state = {
      hidden: props.hidden
    };
    _this.elementMouseDown = null;
    _this.elementTriggeredOpen = null;
    _this.refModal = _react.default.createRef();
    _this.refTransition = _react.default.createRef();
    _this.handleBackdropMouseDown = _this.handleBackdropMouseDown.bind(_assertThisInitialized(_assertThisInitialized(_this)));
    _this.handleBackdropClick = _this.handleBackdropClick.bind(_assertThisInitialized(_assertThisInitialized(_this)));

    if (!props.hidden) {
      _this.elementTriggeredOpen = document.activeElement;
    }

    return _this;
  }

  _createClass(Modal, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      if (!this.state.hidden) {
        this.open();
      }
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps, prevState) {
      if (this.props.hidden !== prevProps.hidden) {
        if (!this.props.hidden) {
          this.elementTriggeredOpen = document.activeElement;
        } // eslint-disable-next-line react/no-did-update-set-state


        this.setState({
          hidden: this.props.hidden
        });
        return;
      }

      if (this.state.hidden && !prevState.hidden) {
        this.close();
      } else if (!this.state.hidden && prevState.hidden) {
        this.open();
      }
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      if (!this.state.hidden) {
        this.close();
      }
    }
  }, {
    key: "open",
    value: function open() {
      this.refModal.current.scrollTop = 0;
      document.body.style.overflow = 'hidden';
    }
  }, {
    key: "close",
    value: function close() {
      var _this2 = this;

      setTimeout(function () {
        document.body.style.overflow = '';
        var returnFocusAfterClose = _this2.props.returnFocusAfterClose;

        if (returnFocusAfterClose && _this2.elementTriggeredOpen !== null) {
          _this2.elementTriggeredOpen.focus();

          _this2.elementTriggeredOpen = null;
        }
      }, 500
      /* Fade delay */
      );
    }
  }, {
    key: "canToggle",
    value: function canToggle() {
      var backdrop = this.props.backdrop;

      if (backdrop === 'static' || backdrop === false) {
        return false;
      }

      var _window$getComputedSt = window.getComputedStyle(this.refTransition.current),
          opacity = _window$getComputedSt.opacity;

      return opacity <= 0.1 || opacity >= 0.9;
    }
  }, {
    key: "handleBackdropClick",
    value: function handleBackdropClick(e) {
      if (e.target !== this.elementMouseDown) {
        return;
      }

      e.stopPropagation();

      if (this.canToggle() && e.target === this.refModal.current) {
        this.props.toggle();
      }
    }
  }, {
    key: "handleBackdropMouseDown",
    value: function handleBackdropMouseDown(e) {
      this.elementMouseDown = e.target;
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          TransitionComponent = _this$props.TransitionComponent,
          ModalWrapperComponent = _this$props.ModalWrapperComponent,
          ModalDialogComponent = _this$props.ModalDialogComponent,
          ModalBackdropComponent = _this$props.ModalBackdropComponent,
          ModalContentComponent = _this$props.ModalContentComponent,
          transitionProps = _this$props.transitionProps,
          wrapperProps = _this$props.wrapperProps,
          dialogProps = _this$props.dialogProps,
          contentProps = _this$props.contentProps,
          backdropProps = _this$props.backdropProps,
          centered = _this$props.centered,
          noRadius = _this$props.noRadius,
          children = _this$props.children,
          backdrop = _this$props.backdrop,
          sm = _this$props.sm,
          lg = _this$props.lg,
          hidden = _this$props.hidden,
          rest = _objectWithoutProperties(_this$props, ["TransitionComponent", "ModalWrapperComponent", "ModalDialogComponent", "ModalBackdropComponent", "ModalContentComponent", "transitionProps", "wrapperProps", "dialogProps", "contentProps", "backdropProps", "centered", "noRadius", "children", "backdrop", "sm", "lg", "hidden"]);

      return _react.default.createElement(TransitionComponent, _extends({
        hidden: this.state.hidden,
        ref: this.refTransition
      }, rest, transitionProps), _react.default.createElement(ModalWrapperComponent, _extends({
        theme: this.props.theme,
        onMouseDown: this.handleBackdropMouseDown,
        onClick: this.handleBackdropClick,
        ref: this.refModal
      }, wrapperProps), _react.default.createElement(ModalDialogComponent, _extends({
        theme: this.props.theme,
        lg: lg,
        sm: sm,
        centered: centered,
        noRadius: noRadius
      }, dialogProps), _react.default.createElement(_ModalContent.ModalContent, _extends({
        theme: this.props.theme
      }, contentProps), children))), !!backdrop && _react.default.createElement(ModalBackdropComponent, _extends({
        theme: this.props.theme,
        backdrop: backdrop
      }, backdropProps)));
    }
  }]);

  return Modal;
}(_react.default.Component);

exports.Modal = Modal;
Modal.defaultProps = {
  theme: _styledConfig.theme,
  TransitionComponent: _styledBaseComponents.Transition,
  ModalWrapperComponent: _ModalWrapper.ModalWrapper,
  ModalDialogComponent: _ModalDialog.ModalDialog,
  ModalBackdropComponent: _ModalBackdrop.ModalBackdrop,
  ModalContentComponent: _ModalContent.ModalContent,
  backdrop: true,
  returnFocusAfterClose: true,
  toggle:
  /* istanbul ignore next */
  function toggle() {}
};
