// styled-form-component/lib/components/FormCheck.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FormCheck = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styledBaseComponents = require("styled-base-components");

var _styledConfig = require("styled-config");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n  position: relative;\n  box-sizing: border-box;\n  & > label {\n    margin-bottom: ", ";\n  };\n  & > input {\n    position: static;\n    margin-top: ", ";\n    margin-right:", ";\n    margin-left: ", ";\n  };\n  ", "\n  ", "\n  ", "\n  ", "\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n    margin-right: ", ";\n  "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n    padding-left: ", ";\n  "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n    align-items: center;\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    display: block;\n  "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n      display: inline-flex;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var display = function display(props) {
  if (props.inline) {
    return (0, _styledComponents.css)(_templateObject());
  }

  return (0, _styledComponents.css)(_templateObject2());
};

var alignItems = function alignItems(props) {
  return props.inline && (0, _styledComponents.css)(_templateObject3());
};

var paddingLeft = function paddingLeft(props) {
  return props.inline && (0, _styledComponents.css)(_templateObject4(), (0, _styledConfig.getPadding)(props, 'formCheck', 'inlineLeft'));
};

var marginRight = function marginRight(props) {
  return props.inline && (0, _styledComponents.css)(_templateObject5(), (0, _styledConfig.getMargin)(props, 'formCheck', 'inlineRight'));
};

var FormCheck = (0, _styledComponents.default)(_styledBaseComponents.Div)(_templateObject6(), function (props) {
  return (0, _styledConfig.getMargin)(props, 'formCheck', 'labelBottom');
}, function (props) {
  return (0, _styledConfig.getMargin)(props, 'formCheck', 'inputTop');
}, function (props) {
  return (0, _styledConfig.getMargin)(props, 'formCheck', 'inputRight');
}, function (props) {
  return (0, _styledConfig.getMargin)(props, 'formCheck', 'inputLeft');
}, function (props) {
  return display(props);
}, function (props) {
  return alignItems(props);
}, function (props) {
  return paddingLeft(props);
}, function (props) {
  return marginRight(props);
});
exports.FormCheck = FormCheck;
FormCheck.defaultProps = {
  theme: _styledConfig.theme
};
